src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -439,6 +439,13 @@ private bool TryRewriteStringConcatenationWithSpanBasedConcat(SyntaxNode syntax,
                     preparedArgs.Add(receiver);
                     continue;
                 }
+                else if (arg.ConstantValueOpt is { IsString: true, StringValue: [char c] })
```

Comment:
I still don't understand why you want to unify lowering towards span-based approach instead of turning constant chars into strings and lower them using normal `string.Concat`. In fact, [this has already worked that way](https://sharplab.io/#v2:EYLgtghglgdgPgAQMwAIECYUGECwAoAb3xRLVQQEYAGFAWQApKaBnAShQF4A+FAcgGNeKANQpmAbnwBfIA==) before all that saga with spans started

Reply:
> If it is not, then I am questioning why bother with span-based approach at all.

The primary benifit of span-based approach is avoiding intermediate allocation during `char` -> `string` transformation. As a tradeoff the size of IL increases. With constant `char` there is no intermediate allocation because we can do transformation at compile time. So by applying span-based approach there we still pay the increased IL size, but the primary benifit is no longer there.
I understand that since you demand this I'll have to deliver it. What I still don't get is why you prefer such little consistency so much over IL size savings (simetimes quite significant as in example below) + little runtime perf boost. I mean, aren't constants always somewhat special from the compiler's point of view: they can be inlined/folded/rearranged just purely based on the fact, that they are known values without evaluation side effects, so it is somewhat natural to expect that emit strategy can differ between non-constant and constant data?

Reply:
> I understand that since you demand this I'll have to deliver it.

That is not a demand, but a suggestion. And it is based on my personal opinion. Other team members might have different opinion. And you can have different opinion as well.

Reply:
> so it is somewhat natural to expect that emit strategy can differ between non-constant and constant data?

Yes, but each case is different.

Reply:
How about the state at commit 48 as a compromise?